plugins/home/src/homePageComponents/WelcomeTitle/timeUtil.ts

```diff
@@ -32,12 +35,33 @@ export function getTimeBasedGreeting(): { language: string; greeting: string } {
       greeting: 'Get some rest',
     };
   }
+
   const timeOfDay = (hour: number): { [language: string]: string } => {
     if (hour < 12) return goodMorning;
     if (hour < 17) return goodAfternoon;
     return goodEvening;
   };
+
   const greetings = timeOfDay(currentHour);
+
+  if (language && Array.isArray(language)) {
+    // Handle the case when language is an array
+    const validLanguages = language.filter(lang => lang && greetings[lang]);
+    if (validLanguages.length > 0) {
+      const greetingsKey = random(validLanguages);
+      return {
+        language: greetingsKey,
+        greeting: greetings[greetingsKey],
+      };
+    }
+  } else if (language && greetings[language]) {
```

Comment:
Would it not simplify things by always using an array? If you just want a single language it would just have one item in the array?